### PR TITLE
CoderTest changes for v0.13.x

### PR DESCRIPTION
**Pull request by shnapz**
Whatever changes were in master re:Coder,CodetTest are manually integrated in this PR.
**upd: more context**
this is a merge of changes from [#4664](https://github.com/spotify/scio/pull/4664):
```
Added more types of checks to should syntax in coder tests
broke down a coder file and coder test spec into multiple files
removed some redundant checks in CoderAssertions
cleaned up the coder spec
```

### CODE REVIEW

**Comment by RustedBones** [comment]
needs to run `scalafix` to remove some imports

**Review by kellen — APPROVED** [review]
Can you add some more context in the PR description? Seems like the most salient change is adding these coder test syntaxes?

File: scio-core/src/main/scala/com/spotify/scio/coders/instances/JodaCoders.scala [hunk 1]
```diff
@@ -49,7 +49,7 @@ object JodaCoders {
     }
 }
 
-final private class JodaDateTimeCoder extends AtomicCoder[DateTime] {
+final private[coders] class JodaDateTimeCoder extends AtomicCoder[DateTime] {
```
[the remark this answers is not in the document]
Comment (shnapz) [reply scio-core/src/main/scala/com/spotify/scio/coders/instances/JodaCoders.scala:52]: the intent of the change - a scope widening to access it from tests :)

File: scio-core/src/main/scala/com/spotify/scio/coders/Coder.scala [hunk 2]
```diff
@@ -681,7 +321,7 @@ private[coders] object CoderStackTrace {
       .currentThread()
       .getStackTrace
       .dropWhile(!_.getClassName.contains(CoderMaterializer.getClass.getName))
-      .take(10)
+      .take(15)
```
[the remark this answers is not in the document]
Comment (shnapz) [reply scio-core/src/main/scala/com/spotify/scio/coders/Coder.scala:324]: stacktrace increased after adding syntax sugar to CoderAssertions, with `10` it cannot reach the CoderMaterializer.getClass.getName in tests, so impossible to verify it works. So totally driven by test case

File: scio-test/src/test/scala/com/spotify/scio/coders/AvroCoderTest.scala [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2021 Spotify AB.
+ * Copyright 2019 Spotify AB.
```
[the remark this answers is not in the document]
Comment (shnapz) [reply scio-test/src/test/scala/com/spotify/scio/coders/AvroCoderTest.scala:2]: This is a new file, but I just copy pasted that header from somewhere :)

**Comment by shnapz** [comment]
@kellen will add more context in the description

**Comment by shnapz** [comment]
@RustedBones `scalafix` is fixed

**Review by RustedBones — APPROVED** [review]

**Pull request merged**